selfie.c

```diff
@@ -2736,8 +2755,10 @@ void createSymbolTableEntry(uint64_t whichTable, uint64_t* string, uint64_t line
   // create entry at head of symbol table
   if (whichTable == GLOBAL_TABLE) {
     setScope(newEntry, REG_GP);
-    setNextEntry(newEntry, global_symbol_table);
-    global_symbol_table = newEntry;
+    
+    // add to hash table
+    setNextEntry(newEntry, getHashEntry(global_hash_symbol_table, string));
+    *(global_hash_symbol_table + getHashValue(string)) = (uint64_t) newEntry;
```

Comment:
You are introducing a new bug here. As you might know, the essence of an hash function is, to map a value of arbitrary size to a smaller value. So as your set of keys increase, also the chance of equal hash values will increase. In this example this would result in overridden symbol table entries. I think closed hashing is the solution to go.